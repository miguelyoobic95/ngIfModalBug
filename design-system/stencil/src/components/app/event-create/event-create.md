---
name: EventCreate
category: Core Components
---

```event-create.html
    <event-create></event-create>
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text inside the label|

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|event-create accent|
|Dark|`dark`|event-create dark|
|Success|`success`|event-create success|
|Danger|`danger`|event-create danger|
|Info|`info`|event-create info|
|Warning|`warning`|event-create warning|
|Gradient Accent|`gradient-accent`|event-create gradient-accent|
|Gradient Dark|`gradient-dark`|event-create gradient-grey|
|Gradient Success|`gradient-success`|event-create gradient-success|
|Gradient Danger|`gradient-danger`|event-create gradient-danger|
|Gradient Info|`gradient-info`|event-create gradient-info|
|Gradient Warning|`gradient-warning`|event-create gradient-warning|

## Events
|Attr|Description|
|---|---|---|
|`yooXXXX`|small label|