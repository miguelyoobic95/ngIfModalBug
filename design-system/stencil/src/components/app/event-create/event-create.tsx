import { Component, Element } from '@stencil/core';

@Component({
    tag: 'yoo-event-create',
    styleUrl: 'event-create.scss',
    scoped: true
})
export class YooEventCreateComponent {

     @Element() host: HTMLStencilElement;

    render(): JSX.Element {
        return (
            <div>EventCreate needs a proper template</div>
        );
    }
}
