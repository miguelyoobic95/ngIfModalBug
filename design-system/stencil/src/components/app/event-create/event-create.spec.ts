import { TestWindow } from '@stencil/core/testing';
import { YooEventCreateComponent } from './event-create';

describe('EventCreateComponent', () => {
    it('should build', () => {
        expect(new YooEventCreateComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooEventCreateComponent],
                html: '<yoo-event-create></yoo-event-create>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {

        let emitEvent, window;
        beforeEach( async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});