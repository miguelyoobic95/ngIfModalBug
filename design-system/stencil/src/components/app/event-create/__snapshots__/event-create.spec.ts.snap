// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCreateComponent Rendering Should render 1`] = `
<yoo-event-create
  class="hydrated"
  data-yoo-event-create-host=""
>
  
  <div
    data-yoo-event-create=""
  >
    EventCreate needs a proper template
  </div>
</yoo-event-create>
`;
